lib/gollum-lib/filter/toc.rb

```diff
@@ -59,6 +68,17 @@ def process(data)
 
   private
 
+  # Generates header in format "h<level>"
+  def make_header_tag(level)
+    raise "Header should be from 1 to 6" unless level.between?(1, 6)
+    "h#{level}"
+  end
+
+  def find_offset(level)
+    tags_before = @missing_headers.select { |number| number < level }
```

Comment:
Not sure yet if I completely understand how this code fixes the bug. What if we have the following situation:

```
h2
h1
...
```

There isn't an h1 before the h2, so presumably this would trigger the bug? But as far as I can see `h1` would not be in `@missing_headers`, because it *is* present in the document. What am I missing?

Reply:
Yes, the PR doesn't handle this situation. I have a wiki and sometimes people prefer using, let's say, h2 headers instead of h1, so they omit h1 headers throughout all the page. `@missing_headers` contain those headers that are missing at all (if it occures at least once, it will reintroduce the bug).

I can try to rewrite my PR so it fixes your case as well, is it necessary?